### DISCUSSION TITLE
Extremely low framerate with non-vsync on Windows ARM64

### DISCUSSION THREAD

**Discussion by hez2010** [ppy/osu#28778] [Q&A]
### Type

Performance

### Bug description

I've got an Windows ARM64 PC (Surface Pro 11), while trying to run osu! x64 version, things seem to work except the framerate is extremely low.
I found that no matter which option I choose (non-vsync options), the framerate won't go higher than 50fps. If I choose VSync it immediately goes higher than 200fps, but the input latency becomes bad and make it hard to play any non-trivial beatmap. 

I'll do some investigation on this, and see if compiling an arm64 version can resolve this issue. 

### Version

2024.625.2

### Logs

-

**Comment by bdach** [comment]
https://github.com/ppy/osu-deploy/pull/170

**Comment by hez2010** [comment]
UPDATE: I managed to build osu with windows arm64, and the issue has been resolved.
Seems like it's a bug related to x64 emulation, and building it with native arm64 resolves the issue.

**Comment by hez2010** [comment]
Sorry. Just checked again and found that the issue still persists. More work need to be done here.

**Comment by smoogipoo** [comment]
WTB: affordable in-stock dev kit from Microsoft :(

**Comment by hez2010** [comment]
I found the root cause for this issue.
It's not related to windows arm64. It's caused by that osu doesn't play well with VRR (variable refresh rate). Turning off VRR resolved the issue.
My display supports VRR at 24~120Hz, with VRR enabled, osu can only render with 24/48 fps (even with unlimited config in osu) on my machine if VSync is turned off.
